src/python/pants/backend/python/tasks/python_binary_create.py

```diff
@@ -126,8 +130,27 @@ def _create_binary(self, binary_tgt, results_dir):
       # Dump everything into the builder's chroot.
       for tgt in source_tgts:
         dump_sources(builder, tgt, self.context.log)
+
+      # Handle locally-built python distribution dependencies.
+      local_dist_req_libs = []
+      built_dists = self.context.products.get_data(BuildLocalPythonDistributions.PYTHON_DISTS)
+      if built_dists:
+        local_dist_req_libs = build_req_libs_provided_by_setup_file(self.context,
+                                                                    built_dists,
+                                                                    self.__class__.__name__)
+      req_tgts = local_dist_req_libs + req_tgts
       dump_requirements(builder, interpreter, req_tgts, self.context.log, binary_tgt.platforms)
 
+      # Dump built python distributions, if any, into builder's chroot.
+      built_dists = self.context.products.get_data(BuildLocalPythonDistributions.PYTHON_DISTS)
```

Comment:
seems like you could just reuse the `built_dists` value from the same call made 9 lines above here?

Reply:
✅